src/nspawn/nspawn-mount.c

```diff
@@ -187,7 +189,7 @@ static int tmpfs_patch_options(
 
         char *buf = NULL;
 
-        if (userns && uid_shift != 0) {
+        if (userns) {
```

Comment:
Because in the case were we do a mount in a new user namespace we can pass the mapped `uid = 0` and the kernel will translate it to the global unmapped uid on the host in the mount options. I can however simply add a bool as arg `bool patch_ids` which can be set to true to force uid/gid patching.